src/v/cloud_storage/partition_manifest.cc

```diff
@@ -345,12 +345,7 @@ bool partition_manifest::advance_start_offset(model::offset new_start_offset) {
 }
 
 std::vector<segment_meta> partition_manifest::replaced_segments() const {
-    std::vector<segment_meta> segments;
-    segments.reserve(_replaced.size());
-    for (const auto& kv : _replaced) {
-        segments.push_back(kv.second);
-    }
-    return segments;
+    return _replaced;
```

Comment:
Maybe this should refer a const ref, now that we're not constructing a fresh object anyway?